src/file.cr

```diff
@@ -3,6 +3,7 @@ require "c/stdio"
 require "c/stdlib"
 require "c/sys/stat"
 require "c/unistd"
+require "path"
```

Comment:
I'm not fond that a core class like `File` have to depend on another class to provide basic functionality like `dirname` and others.

It also creates multiple instances of `Path` which are discarded immediately.

I believe past comment in relation to *rely* was made in the opposite direction: that `Path` relies on `File` and `Dir` classes and not the other way around.